pimd/pim_cmd.c

```diff
@@ -8450,6 +8450,29 @@ DEFPY_HIDDEN (pim_test_sg_keepalive,
 	return CMD_SUCCESS;
 }
 
+DEFPY(interface_ip_pim_allowrp, interface_ip_pim_allowrp_cmd,
+      "[no] ip pim allow-rp [rp-list PLIST]",
+      NO_STR IP_STR PIM_STR
+      "Ignore mismatched RP addresses when processing (*,G) Joins\n"
+      "Specify a prefix-list which the RP address must match in order to be accepted\n"
+      "The prefix-list to check the RP address against\n")
+
+{
+	if (no) {
+		nb_cli_enqueue_change(vty, "./allow-rp", NB_OP_DESTROY, NULL);
+	} else {
+		nb_cli_enqueue_change(vty, "./allow-rp", NB_OP_CREATE, NULL);
+		if (plist) {
+			nb_cli_enqueue_change(vty, "./allow-rp/rp-list",
+					      NB_OP_MODIFY, plist);
+		}
```

Comment:
If the user enters `ip pim allow-rp` after `ip pim allow-rp rp-list plist`, should this command remove the prefix-list? If yes, then you should add the else branch with `NB_OP_DESTROY`.